content/develop/concepts/multipage-apps/widgets.md

```diff
@@ -0,0 +1,57 @@
+---
+title: Working with widgets in multipage apps
+slug: /develop/concepts/multipage-apps/widgets
+description: Understand how widgets interact with pages
+---
+
+# Working with widgets in multipage apps
+
+When you call a widget function in a Streamlit app, Streamlit creates a widget ID and uses it to make your widget stateful. As your users interact with your app and your app reruns, Streamlit keeps track of the widget's value by associating its value to its ID. In particular, a widget's ID depends on the page it's called from. If you define an identical widget on two different pages, then the widget will reset to its default value when you switch pages.
+
+This guide explains several strategies to deal with the behavior if you'd like to have a widget remain stateful across all pages.
+
+## Option 1 (preferred): Call your widget function in your entrypoint file
+
+When you define your multipage app with `st.Page` and `st.navigation`, your entrypoint file becomes a frame of common elements around your pages. When you call a widget function in your entrypoint, it is associated to your entrypoint file instead of a particular page. Since your entrypoint file is executed in every app rerun, any widget in your entrypoint file will remain stateful as your users switch between pages.
```

Comment:
Should we mention here about using session_state to access the widget value within pages?

Reply:
I have an update to this page which links to the bigger article on widget behavior, and to reference it for more detailed information.